helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/OrganizationApplicationService.kt

```diff
@@ -0,0 +1,125 @@
+package de.helfenkannjeder.helfomat.api.organization
+
+import de.helfenkannjeder.helfomat.api.Roles
+import de.helfenkannjeder.helfomat.api.organization.event.OrganizationCreateEventDto
+import de.helfenkannjeder.helfomat.api.organization.event.OrganizationEventAssembler
+import de.helfenkannjeder.helfomat.api.organization.event.OrganizationEventDto
+import de.helfenkannjeder.helfomat.api.organization.event.OrganizationEventDtoAssembler
+import de.helfenkannjeder.helfomat.core.geopoint.BoundingBox
+import de.helfenkannjeder.helfomat.core.geopoint.GeoPoint
+import de.helfenkannjeder.helfomat.core.organization.OrganizationId
+import de.helfenkannjeder.helfomat.core.organization.OrganizationRepository
+import de.helfenkannjeder.helfomat.core.organization.event.ConfirmedChangeOrganizationEvent
+import de.helfenkannjeder.helfomat.core.organization.event.ProposedChangeOrganizationEvent
+import de.helfenkannjeder.helfomat.core.question.QuestionRepository
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.event.EventListener
+import org.springframework.security.access.annotation.Secured
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.stereotype.Service
+
+/**
+ * @author Valentin Zickner
+ */
+@Service
+open class OrganizationApplicationService(
+    private var organizationRepository: OrganizationRepository,
+    private var questionRepository: QuestionRepository,
+    private var applicationEventPublisher: ApplicationEventPublisher) {
+
+    open fun findOrganizationDetails(urlName: String): OrganizationDetailDto? {
+        val organization = organizationRepository.findByUrlName(urlName)
+        val questions = questionRepository.findQuestions()
+        return when (organization) {
```

Comment:
instead of the when you can use `organization?.toDto(questions)`  after you made the methods in `OrganizationAssembler` extension functions.